File: ProcessingSketches/CubicBezierCurve/src/cubicbeziercurve/ControlPoint.java
package cubicbeziercurve;

import processing.core.PApplet;
import toxi.color.TColor;
import toxi.geom.Circle;
import toxi.geom.Vec2D;

public class ControlPoint {
	private Circle _circle;
	private TColor	_color;
	private int		__color;
	private Boolean isSelectable = false;
	
	public ControlPoint( Vec2D aPoint ) {
		setCircle( new Circle( aPoint.x, aPoint.y, 20 )  );
		setHue( 1.0f );		
	}
	
	public void draw() {
		CubicBezierCurve app = CubicBezierCurve.APP;
		app.fill( __color );
		app.ellipse(_circle.x, _circle.y, _circle.getRadius(), _circle.getRadius() );
	}

	///// ACCESSORS
	public Circle getCircle() { return this._circle; }
	public void setCircle(Circle aCircle) { _circle = aCircle; };
	public void setHue( float aHue ) {
		setColor( TColor.newHSV(aHue, 0.6f, 1.0f) );
	}
	
	public void setColor(TColor aColor) { 
		this._color = aColor; 
		__color = this._color.toARGB();
		
		PApplet.println( __color ); 
	}
	public int getColor() { return __color; };
	public float x() { return _circle.x; };
	public float y() { return _circle.y; };
	public void setPosition(Vec2D aPosition ) {
		_circle.x = aPosition.x;
		_circle.y = aPosition.y;
	}
	
	public void setIsSelectable(Boolean isSelectable) { this.isSelectable = isSelectable; }
	public Boolean containsPoint( Vec2D aPoint) { 
		return this._circle.containsPoint( aPoint );
	}
}

File: InterpolationAndSplinesJS/js/curves/ControlPoint.js
/**
 *  A ControlPoint along a curve
 */
window.namespace("Curves");

(function(){
    Curves.ControlPoint = function(x, y, aRadius ){
        this._radius = aRadius || proto.DEFAULT_RADIUS;
        this._circle = new toxi.geom.Circle(x, y, this._radius );

        this._color = random( proto.COLOURS.concat().splice( 1, proto.COLOURS.length-1 ) );
        this._isDraggable = true;
    };

    Curves.ControlPoint.prototype = {
        DEFAULT_RADIUS: 5,
        COLOURS: [ '#69D2E7', '#A7DBD8', '#E0E4CC', '#F38630', '#FA6900', '#FF4E50', '#F9D423' ],

        // @type {Boolean}
        _isDraggable: true,


    /////
        drag: function( touch ) {
            if( !this._isDraggable ) return;

            this._circle.x = touch.x;
            this._circle.y = touch.y;
        },

        /**
         * @param {CanvasRenderingContext2D} ctx
         */
        draw: function( ctx ){
            ctx.beginPath();
            ctx.arc( this._circle.x, this._circle.y, this._circle.radius.x, 0, TWO_PI );
            ctx.fillStyle = this._color;
            ctx.fill();
        },

    ///// ACCESSORS
        getPosition: function(){ return this._circle; },
        setPosition: function( aPosition ) { this._circle.x = aPosition.x; this._circle.y = aPosition.y },

        getColor: function(){ return this._color; },
        setColor: function( aColor ){ this._color = aColor; },

        getRadius: function(){ return this._radius; },

        containsPoint: function(p){ return this._circle.containsPoint(p); }
    }

    var proto = Curves.ControlPoint.prototype;
})();